elastic/kibana
# [Fleet] Only show agent dashboard links if there is more than one non-server agent and if the dashboards exist

## Summary

Closes #161827

These buttons were showing in cloud when there was only the cloud agent added:

<img width="647" alt="Screenshot 2023-08-22 at 16 18 51" src="https://github.com/elastic/kibana/assets/3315046/baaff9b9-17fc-49b5-bad9-980f91a5ae15">

The ingest dashboard links will now only show if:

- the user has one agent that is not part of a policy containing fleet server (I have re-used an existing hook we have to check this here)
- the agent ingest metrics dashaboard exists in the current space

Test Scenarios

1. Setup fleet with ONLY a fleet server connected, the dashboard links should not show
2. Setup fleet with fleet server and another non-fleet server agent, the dashboard links should show
3. With the same setup as test case 2, now create another space, visit fleet in the new space, the links should not show as the dashboards are not installed in the current space.


### Checklist

Delete any items that are not applicable to this PR.
